### PR TITLE
Add brand specific protocol implementation for automotive investigations

### PR DESCRIPTION
**Pull request by polybassa**

### CODE REVIEW

File: scapy/contrib/automotive/bmw/doip.uts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+= Basic Dissect Test
+
+pkt = DoIP(b'\x00\x00\x00\x0a\x00\x01\xf4\x10\x11"3\x11\x11\x11\x11\x11')
+assert pkt.length == 10
+assert pkt.src == 0xf4
+assert pkt.dst == 0x10
+assert pkt.type == 1
+assert pkt[1].service == 17
+assert pkt[2].resetType == 34
```
[the remark this answers is not in the document]
Comment (polybassa) [reply scapy/contrib/automotive/bmw/doip.uts]: Done

**Comment by polybassa** [comment]
@gpotter2 
Thanks for this compliment
Since every car manufactorer does his own interpretation of standards and protocols, I thought that this way would be the most clear one. Some protocols like uds are used by a couple of manufacturers, but are extended individually by every OEM. So this folders could also be a place for brand specific additions to protocols of different OEMs. 
DoIP (Diagnostic over IP) is the same. There is an ISO standart (ISO13400) but some OEMs, like BMW, do there own interpretation. So maybe later, I would like to have the option, to load oem and maybe even model specific protocols through the load_contrib command. (e. g. load_contrib('automotive.bmw.i3.gateway') )

What do you think about that? 

Should I also move my current stuff into this automotive folder ( uds, isotp)?

**Comment by gpotter2** [comment]
> Should I also move my current stuff into this automotive folder ( uds, isotp)?

Might be a good idea

Otherwise it sounds great!

**Comment by polybassa** [comment]
Hi, can this PR get merged?

**Pull request merged**